### PR TITLE
Fix some problems reported by Mirai on Mirai

### PR DESCRIPTION
**Pull request by hermanventer**
## Description

Running MIRAI on itself uncovered some undocumented assumptions. They are now explicit. 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update
- [ ] Additional test coverage

## How Has This Been Tested?
cargo test; validate.sh

### CODE REVIEW

File: checker/src/abstract_value.rs [hunk 1]
```diff
@@ -762,7 +763,7 @@ impl Path {
     pub fn refine_parameters(&self, arguments: &[(Path, AbstractValue)]) -> Path {
         match self {
             Path::LocalVariable { ordinal } if 0 < *ordinal && *ordinal <= arguments.len() => {
-                arguments[ordinal - 1].0.clone()
+                arguments[*ordinal - 1].0.clone()
```
Comment (hermanventer) [inline checker/src/abstract_value.rs]: Not so much an undocumented assumption as a rather subtle call to a virtual function when a simple subtraction is expected.

File: checker/src/constant_domain.rs [hunk 2]
```diff
@@ -274,7 +274,11 @@ impl ConstantDomain {
                 if *val2 == 0 {
                     ConstantDomain::Bottom
                 } else {
-                    ConstantDomain::I128(*val1 / *val2)
+                    let result = val1.checked_div(*val2);
```
Comment (zlandau) [inline checker/src/constant_domain.rs]: ` val1.checked_div(*val2).map(ConstantDomain::I128).map_err(ConstantDomain::Bottom)`

does that work?

Comment (hermanventer) [reply checker/src/constant_domain.rs]: Nope, but I guess you really meant:
`val1.checked_div(*val2).map(ConstantDomain::I128).unwrap_or(ConstantDomain::Bottom)`

**Comment by hermanventer** [comment]
Not easy, no. You'll have to build the latest greatest MIRAI including all the pull requests,, install that in cargo (as per the validate.sh), install Z3, then switch back to before this commit and then run Mirai over itself (as per validate.sh).

**Comment by hermanventer** [comment]
Oh, and you'll have to not use Linux.

**Pull request merged**